webroot/js/directives/sentence-and-translations.dir.js

```diff
@@ -548,19 +548,19 @@
             var data = {
                 value: markupToStored(lang, text)
             };
-
             var i = sentence.transcriptions.findIndex(function(item) {
                 return item.id === transcription.id;
             });
-            sentence.transcriptions.splice(i, 1);
 
+            vm.inProgress = true;
             $http.post(url, data).then(function(result) {
-                sentence.transcriptions.push(result.data.result);
+                transcription = result.data.result;
                 hide('sentence_form');
             }, function(error) {
-                transcription.error = error.statusText;
-                sentence.transcriptions.push(transcription);
+                transcription.error = error.data[0];
```

Comment:
You should check the existence of `error.data[0]` first. This function can be executed in other contexts where no error is return, such as http 500, 401 etc.

Reply:
You are only displaying the first error, however there might be more than one error in the array. [See how the legacy code displays transcription errors](https://github.com/Tatoeba/tatoeba2/blob/dev/src/Template/Transcriptions/view.ctp#L21-L30).